core/src/tests/CMakeLists.txt

```diff
@@ -370,7 +370,10 @@ if(HAVE_EXECINFO_H
   )
 endif() # HAVE_EXECINFO_H ..
 
-if(GMOCK_FOUND AND NOT client-only)
+if(GMOCK_FOUND
+   AND NOT client-only
+   AND NOT ${CMAKE_SYSTEM_NAME} MATCHES "SunOS"
+)
   bareos_add_test(
     sd_reservation LINK_LIBRARIES ${LINK_LIBRARIES} ${GMOCK_LIBRARIES}
   )
```

Comment:
* shouldn't we add the test as disabled, if it cannot be run?
* maybe `bareos_add_test()` should have a parameter `DISABLE_IF` that takes a condition as an argument and disables the test if the condition evaluates to true. I think it would simplify a lot of the CMake-Code around the systemtests.